react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { email: '' };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange() {
```

Comment:
You need to add `event` as a parameter here so you are not accessing the global `event`